File: src/main/kotlin/cz/uhk/ppro/mhjp/resftulgallery/dto/CommentDto.kt
package cz.uhk.ppro.mhjp.resftulgallery.dto

class SubmitCommentDto(
        val content: String = ""
)

open class CommentDataDto(
        val uuid: String,
        val content: String,
        val image: ListImageDataDto,
        val author: ListUserDataDto
) : DataDto()

open class ListCommentDataDto(
        val uuid: String
) : DataDto()

File: src/main/bobril/cz/uhk/ppro/mhjp/gallery/api/dto/commentDto.ts
import {DataDto} from "./baseDto";
import {ListImageDataDto} from "./imageDto";
import {ListUserDataDto} from "./UserDto";

export interface SubmitCommentDto {
     content: string
}

export interface CommentDataDto extends DataDto {
        uuid: string,
        content: string,
        image: ListImageDataDto,
        author: ListUserDataDto
}

export interface ListCommentDataDto extends DataDto {
        uuid: string
}